x-pack/legacy/plugins/alerting/server/alerts_client.ts

```diff
@@ -226,14 +226,29 @@ export class AlertsClient {
   }
 
   public async delete({ id }: { id: string }) {
-    const decryptedAlertSavedObject = await this.encryptedSavedObjectsPlugin.getDecryptedAsInternalUser<
-      RawAlert
-    >('alert', id, { namespace: this.namespace });
+    const [taskIdToRemove, apiKeyToInvalidate] = await Promise.all([
+      this.savedObjectsClient
```

Comment:
It seems like we should - in a future PR - try to fix this code so we only make one SO get call instead of two.  I think, but am not positive, that `getDecryptedAsInternalUser()` returns the un-encrypted properties also returned from `SOC.get()`.  Not worth figuring out for 7.6 tho.

Reply:
Good point, so instead of doing two calls in parallel, you only do `SOC.get()` if ever `getDecryptedAsInternalUser()` throws an error (in series)?

Reply:
I'd have to look closer at the impact, but it just doesn't seem right that we're doing two SO lookups, back-to-back.  Used to be serially, I think, now it's in parallel, which is better perf-wise for this API, but still hits ES twice, which seems unfortunate, and hopefully, not actually required.  If that seems possible, let's open another issue to investigate later.  It would be a performance improvement ...

Reply:
I have created #56777 to fix this.